components/PendingRequestsList.tsx

```diff
@@ -35,11 +39,97 @@ export const PendingRequestsList: React.FC = React.memo(() => {
   const primaryTextColor = useThemeColor({}, 'textPrimary');
   const secondaryTextColor = useThemeColor({}, 'textSecondary');
 
+  // Type guards for safe metadata access
+  const hasExpiresAt = (metadata: unknown): metadata is { expiresAt: number } => {
+    return (
+      typeof metadata === 'object' &&
+      metadata !== null &&
+      'expiresAt' in metadata &&
+      typeof (metadata as Record<string, unknown>).expiresAt === 'number'
```

Comment:
If i remember correctly the timestamp is a bigint because it's mapped from a rust `u64`. Does it work with this code still?

This also applies to all the other places below